pkg/controller/controller_test.go

```diff
@@ -3156,3 +3129,257 @@ func TestProvisionFromPVC(t *testing.T) {
+					func(pv *v1.PersistentVolume) (*v1.PersistentVolume, error) {
+						newPV := pv.DeepCopy()
+						if newPV.Annotations == nil {
+							newPV.Annotations = map[string]string{}
+						}
+						newPV.Annotations[translatedAnnKey] = "foo"
+						return newPV, nil
+					},
+				).AnyTimes()
+			}
+
+			// Set an expectation that the Create should be called
+			if tc.expectCreate {
+				expectParams := map[string]string{"fstype": "ext3"} // Default
+				if tc.expectTranslation {
+					// If translation is expected we check that the CreateVolume
+					// is called on the expected volume with a translated param
+					expectParams[translatedAnnKey] = "foo"
+				}
+				controllerServer.EXPECT().CreateVolume(gomock.Any(),
+					&csi.CreateVolumeRequest{
+						Name:               "test-testi",
+						Parameters:         expectParams,
+						VolumeCapabilities: nil,
+						CapacityRange: &csi.CapacityRange{
+							RequiredBytes: int64(requestBytes),
+						},
+					}).Return(
+					&csi.CreateVolumeResponse{
+						Volume: &csi.Volume{
+							CapacityBytes: requestBytes,
+							VolumeId:      "test-volume-id",
+						},
+					}, nil).Times(1)
+			}
+
+			// Make a Provision call
+			volOpts := controller.ProvisionOptions{
+				StorageClass: &storagev1.StorageClass{
+					Provisioner:   tc.provisioner,
+					Parameters:    map[string]string{"fstype": "ext3"},
+					ReclaimPolicy: &deletePolicy,
+				},
+				PVName: "test-name",
+				PVC:    createFakePVCWithAnnotation(tc.annotation, requestBytes),
+			}
+
+			pv, state, err := csiProvisioner.(controller.ProvisionerExt).ProvisionExt(volOpts)
+			if tc.expectErr && err == nil {
+				t.Errorf("Expected error, got none")
+			}
+			if err != nil {
+				if !tc.expectErr {
+					t.Errorf("got error: %v", err)
+				}
+				return
+			}
+
+			if controller.ProvisioningFinished != state {
+				t.Errorf("expected ProvisioningState %s, got %s", controller.ProvisioningFinished, state)
+			}
+
+			if tc.expectTranslation {
+				if _, ok := pv.Annotations[translatedAnnKey]; !ok {
+					t.Errorf("got no translated annotation %s on the pv, expected one", translatedAnnKey)
+				}
+			}
+		})
+
+	}
+}
+
+func createFakePVCWithAnnotation(ann map[string]string, requestBytes int64) *v1.PersistentVolumeClaim {
+	return &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:         "testid",
+			Name:        "fake-pvc",
+			Annotations: ann,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			Selector: nil, // Provisioner doesn't support selector
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): resource.MustParse(strconv.FormatInt(requestBytes, 10)),
+				},
+			},
+		},
+	}
+}
+
+func TestDeleteMigration(t *testing.T) {
+	var (
+		translatedHandle = "translated-handle"
+		normalHandle     = "no-translation-handle"
+	)
+
+	testCases := []struct {
+		name           string
+		pv             *v1.PersistentVolume
+		translateToCSI bool
```

Comment:
nit: would it make sense to rename this to `expectTranslation` as per the previous test?